org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/FeatureCallCompiler.java

```diff
@@ -562,19 +562,32 @@ protected void _toJavaExpression(XAbstractFeatureCall call, ITreeAppendable b) {
 		}
 	}
 
+	private static boolean isConstantExpression(JvmAnnotationReference reference) {
+		for (final JvmAnnotationValue annotationValue: reference.getValues()) {
+			if ("constantExpression".equals(annotationValue.getValueName())) {
+				return ((JvmBooleanAnnotationValue) annotationValue).getValues().get(0).booleanValue();
+			}
+		}
+		return false;
+	}
+
 	protected void featureCalltoJavaExpression(final XAbstractFeatureCall call, ITreeAppendable b, boolean isExpressionContext) {
 		if (call instanceof XAssignment) {
 			assignmentToJavaExpression((XAssignment) call, b, isExpressionContext);
 		} else {
 			if (needMultiAssignment(call)) {
 				appendLeftOperand(call, b, isExpressionContext).append(" = ");
 			}
-			boolean hasReceiver = appendReceiver(call, b, isExpressionContext);
-			if (hasReceiver) {
-				b.append(".");
-				b = appendTypeArguments(call, b);
+			ITreeAppendable c = b;
+			final JvmAnnotationReference annotationRef = this.expressionHelper.findInlineAnnotation(call);
+			if (annotationRef == null || !isConstantExpression(annotationRef)) {
+				boolean hasReceiver = appendReceiver(call, b, isExpressionContext);
+				if (hasReceiver) {
+					b.append(".");
+					c = appendTypeArguments(call, b);
+				}
 			}
-			appendFeatureCall(call, b);
+			appendFeatureCall(call, c);
```

Comment:
Because the appendable is overriden at line 587 for fixing the original issue.
I will push a new version without a local variable.